app/frontend/src/components/Navigation/Navigation.tsx

```diff
@@ -115,13 +117,22 @@ export default function Navigation() {
   const authClasses = useAuthStyles();
   const authenticated = useAuthContext().authState.authenticated;
   const [open, setOpen] = React.useState(false);
+  const { data } = useNotifications();
+    
+  let notificationNumber = data?.unseenMessageCount;
 
   const drawerItems = (
     <div>
       <List>
         {menu.map(({ name, route }) => (
           <ListItem button key={name}>
-            <NavButton route={route} label={name} labelVariant="h2" />
+            {route === messagesRoute ? (
+              <NotificationBadge count={notificationNumber}>
+                <NavButton route={route} label={name} labelVariant="h2" />
+              </NotificationBadge>
+            ) : (
+              <NavButton route={route} label={name} labelVariant="h2" />
+            )}
```

Comment:
This doesn't seem ideal but I can't think of a better way. Later we will have a dedicated notifications dropdown like facebook so I think it's fine for now.

Reply:
Dedicated notifications seems a good idea, will the message tab be remaining in the navbar then?